lib/cn/cn.c

```diff
@@ -933,58 +935,65 @@ cn_open(
     }
 
     ctx.cn = cn;
-    ctx.nodemap = map_create(cn->cp->fanout);
+    ctx.nodemap = map_create(CN_FANOUT_MAX);
     if (ev(!ctx.nodemap)) {
         err = merr(ENOMEM);
         goto err_exit;
     }
 
     map_insert_ptr(ctx.nodemap, 0, cn->cn_tree->ct_root);
 
-    /* [HSE_REVISIT] Delete this block once we have splits. cn_kvset_cb() will populate the nodemap,
-     * and then for each node in the map, fetch the max key and call route_map_insert() with it.
-     */
-    {
-        struct ekey_generator *egen;
+    err = cndb_cn_instantiate(cndb, cnid, &ctx, (void *)cn_kvset_cb);
+    if (ev(err))
+        goto err_exit;
 
-        egen = ekgen_create(kvs_name, cn->cp);
-        assert(egen);
+    if (cn->cn_tree->ct_route_map) {
+        char kbuf[HSE_KVS_KEY_LEN_MAX];
+        struct cn_tree_node *tn, *next;
+        struct route_node *rn;
+        uint klen;
```

Comment:
You can't get here without a routemap.  No need for `if (cn->cn_tree->ct_route_map)`.

Reply:
Yeah, I was following existing checks, was wondering in what case we wouldn't have a route map.
Actually, I think there is this once confounding case where we have a partially set up tree, lemme find it...

Reply:
In cn_make() will build a half-baked cn tree then tear it down right away.  In that case we don't bother creating the route map.  But it should never be the case where we're in cn_open() and don't have a route map.

Reply:
Ya know, I'll rework this to remove the check for the route map and then I'll probably have to spend three hours fixing some stupid unit test...

Reply:
yay, unit tests passed!

Reply:
Done.

Reply:
> fixing some stupid unit test

I was afraid that's why the check was there in the first place. :)